apps/server/src/modules/idp-console/idp-console.module.ts

```diff
@@ -0,0 +1,81 @@
+import { Module } from '@nestjs/common';
+import { ConsoleModule } from 'nestjs-console';
+import { CqrsModule } from '@nestjs/cqrs';
+import { ConsoleWriterModule } from '@infra/console';
+import { ConfigModule } from '@nestjs/config';
+import { HttpModule, HttpService } from '@nestjs/axios';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Logger, LoggerModule } from '@src/core/logger';
+import { MikroOrmModule } from '@mikro-orm/nestjs';
+import { DB_PASSWORD, DB_URL, DB_USERNAME } from '@src/config';
+import { SchulconnexRestClient } from '@infra/schulconnex-client';
+import { AccountModule } from '@modules/account';
+import { OauthAdapterService } from '@modules/oauth';
+import { defaultMikroOrmOptions } from '@modules/server';
+import { ALL_ENTITIES } from '@shared/domain/entity';
+import { UserService } from '@modules/user';
+import { UserRepo } from '@shared/repo';
+import { UserDORepo } from '@shared/repo/user/user-do.repo';
+import { RoleModule } from '@modules/role';
+import { IdentityManagementModule } from '@infra/identity-management';
+import { RegistrationPinModule } from '@modules/registration-pin';
+import {
+	SynchronizationEntity,
+	SynchronizationRepo,
+	SynchronizationService,
+	SynchronizationUc,
+} from '@modules/synchronization';
+import { IdpSyncConsole } from './idp-sync-console';
+
+@Module({
+	imports: [
+		ConfigModule.forRoot({ isGlobal: true }),
+		ConsoleModule,
+		ConsoleWriterModule,
+		LoggerModule,
+		CqrsModule,
+		MikroOrmModule.forRoot({
+			...defaultMikroOrmOptions,
+			type: 'mongo',
+			clientUrl: DB_URL,
+			password: DB_PASSWORD,
+			user: DB_USERNAME,
+			allowGlobalContext: true,
+			entities: [...ALL_ENTITIES, SynchronizationEntity],
+			debug: true,
+		}),
+		IdentityManagementModule,
+		HttpModule,
+		AccountModule,
+		RegistrationPinModule,
+		RoleModule,
+	],
+	providers: [
+		SynchronizationRepo,
```

Comment:
the Synchronisation Module also is another module, import it instead of providing its contents.

Reply:
I've done that, but only partly (see https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859/commits/e742fbb95a9912633f496d02e83cb1e1bc872594 ) - I had to leave `SynchronizationUc` as exporting it from the `SynchronizationModule` would require to provide `SchulconnexRestClient` to it and I want to do this only in the `IdpConsoleModule`.

Reply:
Ok I'll move `SynchronizationUc` to the `idp-console` module as it'll fix this architectural issue as well https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859#discussion_r1556022329 .

Reply:
Done in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859/commits/c79e0170a2cf8e9cd4f1de410d4373565276b5af .